src/Provider/DatabaseMenuProvider.php

```diff
@@ -24,20 +20,20 @@ class DatabaseMenuProvider implements MenuProviderInterface
     protected $builder = null;
 
     /**
-     * @var ContainerInterface
+     * @var RequestStack
      */
-    protected $container;
+    protected $requestStack;
 
     /**
-     * Constructor
-     *
-     * @param Builder $builder
-     * @param ContainerInterface $container
+     * @var MatcherInterface
      */
-    public function __construct(BuilderInterface $builder, ContainerInterface $container)
+    protected $matcher;
+
+    public function __construct(BuilderInterface $builder, RequestStack $requestStack, MatcherInterface $matcher = null)
```

Comment:
Kan het eigenlijk voorkomen dat de `$matcher` null is? Wanneer zou dat bijv. kunnen? Als je dat wilt aanbieden moet ook je definitie iets anders denk ik, bijv `<argument type="service" id="knp_menu.matcher" on-invalid="null" />`